client/ayon_applications/manager.py

```diff
@@ -524,26 +535,60 @@ def manager(self) -> ApplicationManager:
         return self.application.manager
 
     def _run_process(self) -> subprocess.Popen:
-        # Windows and MacOS have easier process start
+        """Run the process with the given launch arguments and keyword args.
+
+        This method will handle the process differently based on the platform
+        it is running on. It will create a temporary file for output on
+        Windows and macos, while on Linux it will use a mid-process to launch
+        the application with the provided arguments and environment variables.
+
+        It will pass file paths to temporary files to the mid-process where
+        the process output and pid will be stored.
+
+        Returns:
+            subprocess.Popen: The process object created by Popen.
+
+        """
+        # Windows and macOS have easier process start
         low_platform = platform.system().lower()
         if low_platform in ("windows", "darwin"):
-            return subprocess.Popen(self.launch_args, **self.kwargs)
-
-        # Linux uses mid process
-        # - it is possible that the mid process executable is not
+            return self._execute_with_stdout()
+        # Linux uses mid-process
+        # - it is possible that the mid-process executable is not
         #   available for this version of AYON in that case use standard
         #   launch
         launch_args = get_linux_launcher_args()
         if launch_args is None:
             return subprocess.Popen(self.launch_args, **self.kwargs)
 
-        # Prepare data that will be passed to midprocess
+        # Prepare data that will be passed to mid-process
         # - store arguments to a json and pass path to json as last argument
         # - pass environments to set
         app_env = self.kwargs.pop("env", {})
+        # create temporary file path passed to mid-process
+        with tempfile.NamedTemporaryFile(
+            mode="w",
+            prefix=f"ayon_{self.application.host_name}_output_",
+            suffix=".txt",
+            delete=False
+        ) as temp_file:
+            output_file = temp_file.name
+        # create temporary file to read back pid
+        with tempfile.NamedTemporaryFile(
+            mode="w", prefix="ayon_pid_", suffix=".txt", delete=False
+        ) as pid_temp_file:
+            pid_file = pid_temp_file.name
+
         json_data = {
```

Comment:
What is `site_id` and `output` used for? Is there any reason to send `pid_file` to the subprocess?